js/modules/types/attachment/migrate_data_to_file_system.js

```diff
@@ -0,0 +1,37 @@
+const isArrayBuffer = require('lodash/isArrayBuffer');
+const isFunction = require('lodash/isFunction');
+const isUndefined = require('lodash/isUndefined');
+
+
+// type Context :: {
+//   writeAttachmentData :: ArrayBuffer -> Promise (IO Path)
+// }
+//
+//      migrateDataToFileSystem :: Attachment ->
+//                                 Context ->
+//                                 Promise Attachment
+exports.migrateDataToFileSystem = async (attachment, { writeAttachmentData } = {}) => {
+  if (!isFunction(writeAttachmentData)) {
+    throw new TypeError('`writeAttachmentData` must be a function');
+  }
+
+  const { data } = attachment;
+  const hasData = !isUndefined(data);
+  const shouldSkipSchemaUpgrade = !hasData;
+  if (shouldSkipSchemaUpgrade) {
+    console.log('WARNING: `attachment.data` is `undefined`');
+    return attachment;
+  }
+
+  const isValidData = isArrayBuffer(data);
+  if (!isValidData) {
+    throw new TypeError('Expected `attachment.data` to be an array buffer;' +
+      ` got: ${typeof attachment.data}`);
+  }
+
+  const path = await writeAttachmentData(data);
+
+  const attachmentWithoutData = Object.assign({}, attachment, { path });
+  delete attachmentWithoutData.data;
+  return attachmentWithoutData;
```

Comment:
Grr, forgot to use `omit` again. Can fix it before merging.

Reply:
It's not a big deal, I'm just honestly surprised that it's not second-nature for you! :0)

Reply:
I didn’t start using Lodash until recently, that’s why. Deleting fields from a data type is also not something you do in statically typed languages 😉

Reply:
Fixed it. Thanks for the tip. This is one example where yes, we could say that comment is not worth it but I appreciate you having made it and I learned something new 😄